tests/test_execution.py

```diff
@@ -1333,7 +1334,50 @@ def clear_state_program(self) -> bool:
                 return True
 
     # expected error message here is just an example, update test once a better one is available
-    import puya.errors
-
     with pytest.raises(puya.errors.CodeError, match="Invalid use of a type union"):
         harness.deploy_from_closure(test)
+
+
+def test_mutable_ref_param_reassignment_fails(harness: _TestHarness) -> None:
```

Comment:
These would be better as part of the expected_errors.test suite?

Reply:
Because I thought (wrongly) that those tests didn't lower to IR, where this error is generated.

Reply:
I've moved them as I suggested